src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -190,6 +190,12 @@
   public static final String PARTITION_FIELD_NAME_DEFAULT = "";
   public static final String PARTITION_FIELD_NAME_DISPLAY = "Partition Field Name";
 
+  public static final String PARTITION_TIME_FIELD_NAME_CONFIG = "partition.timefield.name";
+  private static final String PARTITION_TIME_FIELD_NAME_DOC =
+          "The timestamp field name of SinkRecord when TimeBasedPartitioner is used, "
+          + "current time will be used on empty";
+  public static final String PARTITION_TIME_FIELD_NAME_DEFAULT = "";
```

Comment:
My first attempt on eliminating default value failed, looked into `ConfigDef.parse()`, can't find a way to express "this config key is required only when partitioner.class is `TimeBasedPartitioner`". Suggestions are welcomed.